### PR TITLE
use datetime as version so the migration bot can work properly

### PR DESCRIPTION
**Pull request by CJ-Wright**
@isuruf @msarahan @jakirkham @beckermr

I think this is a resonable resolution to some of the issues the bot has been running into when issuing migrations.
This mostly solves the UX issue where the bot's PRs must be rebased/merged with upstream when multiple migrations are being merged.

Closes https://github.com/conda-forge/conda-smithy/issues/1270

xref: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/39

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{% set version = "2020.03.06" %}
+{% set version = datetime.datetime.utcnow().strftime('%Y.%m.%d.%H.%M.%S') %}
```
Comment (isuruf) [inline recipe/meta.yaml]: Do we need seconds?

**Comment by isuruf** [comment]
+1 for using datetime.

File: recipe/meta.yaml (hunk 1 above)
Comment (CJ-Wright) [reply recipe/meta.yaml]: I'm worried that if two PRs get merged fast enough + the inconsistency of the CI pushing the builds through could have two builds in the same minute.

**Comment by beckermr** [comment]
Agreed. LGTM!

**Comment by CJ-Wright** [comment]
After this is in I'll ask the bot to rerun the outstanding PRs so that the version is correct (this should also ping the associated teams). Then we can get most of this on the road, the bot is still missing the impact analysis for how many packages will get a PR.

File: recipe/meta.yaml (hunk 1 above)
Comment (isuruf) [reply recipe/meta.yaml]: :smile: ok

**Comment by beckermr** [comment]
We’ll need a pr to adjust the bot code first. It tried to bump the version right now.

**Comment by beckermr** [comment]
Also I merged a pr with the version set to tomorrow. So we’ll have to wait for that too

**Comment by CJ-Wright** [comment]
https://github.com/regro/cf-scripts/pull/895

**Comment by jakirkham** [comment]
LGTM. Thanks CJ! 😀

**Comment by CJ-Wright** [comment]
I think this is ready to go. I'll ask the bot for a rerun of the other PRs that are ready to go.

**Comment by CJ-Wright** [comment]
The build seems unhappy? https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=135242&view=logs&jobId=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&j=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&t=ec8d466d-e3b4-5115-4f06-222398f91e6c

**Comment by isuruf** [comment]
Ah, yes. `upload_package` computes the name and version by rendering, but the version is different because the timestamp has changed.

**Comment by CJ-Wright** [comment]
:man_facepalming:

**Comment by beckermr** [comment]
Ohhhh no!

**Comment by CJ-Wright** [comment]
Maybe this is minute stable, but not second stable.

**Comment by beckermr** [comment]
If we only record hours it will work, but that doesn't help much.

**Comment by isuruf** [comment]
We should change `upload_package` to just upload all built artifacts.

**Comment by beckermr** [comment]
Usually a build is a few minutes @CJ-Wright 

good idea @isuruf

**Comment by CJ-Wright** [comment]
@isuruf can you provide more info about where this needs to be fixed?

**Comment by beckermr** [comment]
here I think: https://github.com/conda-forge/conda-forge-ci-setup-feedstock/blob/master/recipe/conda_forge_ci_setup/upload_or_check_non_existence.py#L111

**Comment by CJ-Wright** [comment]
yes but where are the actual file paths? I don't know if we can use `conda_build.api.get_output_file_paths`

**Comment by jakirkham** [comment]
What were we using before?

**Comment by beckermr** [comment]
IDK exactly, but I see lines in the condarc like this

```
conda-build:
  root-dir: /home/conda/feedstock_root/build_artifacts
```

and envvar exports like this

```
 echo 'export CONDA_BLD_PATH='\''/home/conda/feedstock_root/build_artifacts'\'''
```

**Comment by isuruf** [comment]
`conda_build.config.croot` + subdir and noarch

**Comment by beckermr** [comment]
How is that path set from the env @isuruf ?

**Comment by beckermr** [comment]
https://github.com/conda/conda-build/blob/3072a08f4826b044b32ecb69c65a4c7a25435bed/conda_build/config.py#L402

**Pull request merged**